openslides/core/views.py

```diff
@@ -353,3 +359,19 @@ def check_permissions(self, request):
         if (self.action in ('create', 'update', 'destroy') and
                 not request.user.has_perm('core.can_manage_tags')):
             self.permission_denied(request)
+
+
+class UrlPatternsView(utils_views.APIView):
+    """
+    Returns a dictonary with all url patterns as json.
+    """
+    URL_KWARGS_REGEX = re.compile(r'%\((\w*)\)s')
+
+    def get_context_data(self, **context):
+        result = {}
+        url_dict = get_resolver(None).reverse_dict
+        for url_pattern in filter(lambda key: isinstance(key, str), url_dict.keys()):
+            url = url_dict[url_pattern][0][0][0]
```

Comment:
Django allows multiple urls for one pattern_name. This makes sense for patterns with different arguments. We use this feature for agenda_speaker_delete, see https://github.com/OpenSlides/OpenSlides/blob/master/openslides/agenda/urls.py#L58 (this url will be removed in 2.0)

But this feature is not supported by the angular ui router. So the pattern_name in this function has to be unique.